definitions/npm/express_v4.x.x/flow_v0.25.x-/express_v4.x.x.js

```diff
@@ -12,7 +12,7 @@ declare module 'express' {
   }
   declare class Request extends http$IncomingMessage mixins RequestResponseBase {
     baseUrl: string;
-    body: mixed;
+    body: any;
```

Comment:
@vaukalak: TBH I would probably never advocate for mutating an internal thing in that way as a side-effect-based contract, but if it's a common pattern then we should probably handle it.

All this said, it occurs to me that this is controversial because different developers/use-cases demand/expect certain levels of strictness/looseness. What if we made the `body` type a type parameter and asked developers to opt-in to their stated body type?

```js
declare class Request<BodyT> {
  baseUrl: string;
  body: BodyT;
  cookies: {[cookie: string]: string};
  ...etc...
}
```

Then to use:

```js
app.post('/', (req, res) => {
  console.log(`req.body: ${JSON.stringify(req.body)}`);
  console.log(`typeof req.body: ${typeof req.body}`); // string
  res.send('kuku');
});
```

^^ works leaving the `BodyT` type as just the "`empty`" type since Flow doesn't know what it is (`empty` can flow into anything, but nothing can flow into it..a bit of a strange type, but it models Flow's understanding of the type correctly).

And a user who wishes to be specific can simply annotate the `res` parameter:

```js
type MyResponseT = /* ... */;
app.post('/', (req: Response<MyResponseT>, res) => {
  console.log(`req.body: ${JSON.stringify(req.body)}`);
  console.log(`typeof req.body: ${typeof req.body}`); // string
  res.send('kuku');
});
```

What do you think? (cc also: @bassettsj )